config/options.go

```diff
@@ -557,11 +557,24 @@ func (o *Options) Validate() error {
 		o.Certificates = append(o.Certificates, *cert)
 	}
 
-	for _, c := range o.CertificateFiles {
-		cert, err := cryptutil.CertificateFromFile(c.CertFile, c.KeyFile)
-		if err != nil {
-			return fmt.Errorf("config: bad cert file %w", err)
+	for i, c := range o.CertificateFiles {
+		fileRefErr := error(nil)
+		cert, base64Err := cryptutil.CertificateFromBase64(c.CertFile, c.KeyFile)
+		
+		if base64Err != nil {
```

Comment:
I don't think this need to be complicated like this. Just use one error like your original version, and said that "either a base64 or a file reference" in error message.